pkg/karmadactl/join/join.go

```diff
@@ -196,11 +197,21 @@ func (j *CommandJoinOption) Run(f cmdutil.Factory) error {
 	return j.RunJoinCluster(controlPlaneRestConfig, clusterConfig)
 }
 
+var controlPlaneKubeClientBuilder = func(controlPlaneRestConfig *rest.Config) kubeclient.Interface {
+	return kubeclient.NewForConfigOrDie(controlPlaneRestConfig)
+}
+var karmadaClientBuilder = func(controlPlaneRestConfig *rest.Config) karmadaclientset.Interface {
+	return karmadaclientset.NewForConfigOrDie(controlPlaneRestConfig)
+}
+var clusterKubeClientBuilder = func(clusterConfig *rest.Config) kubeclient.Interface {
+	return kubeclient.NewForConfigOrDie(clusterConfig)
+}
```

Comment:
The implementation of controlPlaneKubeClientBuilder is the same as that of clusterKubeClientBuilder.

Reply:
I think we can share a more common method variable naming.

Reply:
@XiShanYongYe-Chang We could unify them under a single method-naming variable, but doing so would prevent us from distinguishing between the `controlPlaneKubeClient` and `clusterKubeClient` in the test cases. Since we don't use the `*rest.Config` in the test cases, we instead mock the functions that return those types of clients. By using a single method-naming variable, we would lose the flexibility to mock and test the clients separately, reducing control in our test cases.